org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java

```diff
@@ -164,6 +164,7 @@ public static void initializeJavaCoreOptions() {
 		// workaround for https://github.com/redhat-developer/vscode-java/issues/718
 		javaCoreOptions.put(JavaCore.CORE_CIRCULAR_CLASSPATH, JavaCore.WARNING);
 		javaCoreOptions.put(JavaCore.COMPILER_IGNORE_UNNAMED_MODULE_FOR_SPLIT_PACKAGE, JavaCore.ENABLED);
+		javaCoreOptions.put(JavaCore.COMPILER_PB_SYNTACTIC_NULL_ANALYSIS_FOR_FIELDS, JavaCore.ENABLED);
```

Comment:
This is exactly where I thought of putting it too. Then I realized null analysis is one of the few settings we enable in a very special way. We don't just outright enable it by default. We do some more complicated logic to first figure out if the needed libraries are on the classpath. Then we enable/disable it as required.